data_log_colab.ipynb

```diff
@@ -698,20 +664,84 @@
     "X_test_sc = scaler.transform(X_test)"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Support Vector Machine multi-step regressor"
+   ]
+  },
   {
    "cell_type": "code",
    "execution_count": null,
-   "metadata": {
-    "colab": {},
-    "colab_type": "code",
-    "id": "X7CR3T7Gdsa5"
-   },
+   "metadata": {},
    "outputs": [],
    "source": [
-    "BATCH_SIZE = 256  # batch size for training\n",
-    "EPOCHS = 400      # epochs for training\n",
-    "WAIT = 50         # patience for early stopping\n",
-    "LR = 1e-3         # initial learning rate"
+    "# Single-step ahead time-series regression\n",
+    "# Randomized grid search with cross-validation\n",
+    "parameters = {'C':stats.uniform(loc=1e-4, scale=1e5),\n",
+    "              'epsilon':stats.uniform(loc=1., scale=1e4),\n",
+    "              'gamma':['auto', 'scale']}\n",
+    "svr = RandomizedSearchCV(estimator=SVR(kernel='rbf'), \n",
+    "                         param_distributions=parameters, n_iter=1000, \n",
+    "                         cv=TimeSeriesSplit(n_splits=3), \n",
+    "                         scoring='neg_mean_squared_error', \n",
+    "                         refit=True, n_jobs=-1, verbose=1)\n",
+    "# Fit on training data using single-step time lag target values\n",
+    "svr.fit(X_train, y_train.values[:,0])  # do not use scaled data\n",
+    "print(svr.best_params_)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Using best hyper-parameters from the single-step ahead regression\n",
+    "multi_svr = MultiOutputRegressor(estimator=SVR(kernel='rbf', **svr.best_params_), n_jobs=-1)\n",
+    "multi_svr.fit(X_train.values, y_train.values)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# A multi-step model that arranges regressions into a chain. Each model makes a prediction\n",
+    "# in the order specified by the chain (i.e. order of columns in the target matrix) using\n",
+    "# all of the available features provided to the model plus the predictions of models that\n",
+    "# are earlier in the chain. Order of columns is arranged by time-lags. Base model is SVM!\n",
+    "chain_svr = RegressorChain(base_estimator=SVR(kernel='rbf', **svr.best_params_))\n",
+    "chain_svr.fit(X_train.values, y_train.values)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## DecisionTree multi-step regressor"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# DecisionTreeRegressor supports multi-step output out-of-the-box!\n",
+    "# Grid search with cross-validation\n",
+    "parameters = [{'criterion':['mse', 'mae'],\n",
```

Comment:
Maybe we can remove this bracket as well.

Reply:
Yes, we can try it.